src/main/scala/org/downtowndailybread/bethsaida/request/ServiceRequest.scala

```diff
@@ -0,0 +1,208 @@
+package org.downtowndailybread.bethsaida.request
+
+import java.sql.{Connection, ResultSet}
+import java.util.UUID
+
+import org.downtowndailybread.bethsaida.exception.service.{ScheduleNotFoundException, ServiceNotFoundException}
+import org.downtowndailybread.bethsaida.model.{InternalUser, Schedule, ScheduleDetail, Service, ServiceAttributes, ServiceType}
+import org.downtowndailybread.bethsaida.request.util.{BaseRequest, DatabaseRequest}
+import org.downtowndailybread.bethsaida.service.UUIDProvider
+
+class ServiceRequest(val conn: Connection) extends BaseRequest with DatabaseRequest with UUIDProvider {
+
+  def rsConverter(rs: ResultSet): Service = {
+    Service(
+      parseUUID(rs.getString("id")),
+      ServiceAttributes(
+        rs.getString("name"),
+        ServiceType.withName(rs.getString("type"))
+      ),
+      if(rs.getString("schedule_id") == null) {
+        Seq()
+      } else {
+        Seq(Schedule(
+          parseUUID(rs.getString("schedule_id")),
+          ScheduleDetail(
+            rs.getString("rrule"),
+            rs.getBoolean("enabled")
+          )
+        ))
+      }
+    )
+  }
+
+  def getAllServices(): Seq[Service] = serviceGetter(None)
+
+  def getService(id: UUID): Service = serviceGetter(Some(id)).toList match {
```

Comment:
If we are only expecting one, would it make sense to get the headOption? Otherwise, my first thing looking at this is that it can get multiple and would throw a not found exception

Reply:
Yeah, that was sort of on purpose. It shouldn't be possible for multiple responses to come back, and I want it to be an error. Technically, it has no idea how to find the right one ;)